contracts/BaseStrategy.sol

```diff
@@ -602,21 +602,21 @@ abstract contract BaseStrategy {
      * transferring any reserve or LP tokens, CDPs, or other tokens or stores of
      * value.
      */
-    function prepareMigration(address _newStrategy) internal virtual;
+    function prepareMigration(address _newStrategy) internal virtual returns (bool _success);
 
     /**
      * @notice
      *  Transfers all `want` from this Strategy to `_newStrategy`.
      *
-     *  This may only be called by governance or the Vault.
+     *  This may only be called by governance or the Vault. The Strategy must explicitly signal success of migration prerequisites before balance is transferred to `_newStrategy`
      * @dev
      *  The new Strategy's Vault must be the same as this Strategy's Vault.
      * @param _newStrategy The Strategy to migrate to.
      */
     function migrate(address _newStrategy) external {
         require(msg.sender == address(vault) || msg.sender == governance());
         require(BaseStrategy(_newStrategy).vault() == vault);
-        prepareMigration(_newStrategy);
+        require(prepareMigration(_newStrategy));
```

Comment:
what's the worst-case scenario? 
if the migration fails, want might fail.

Reply:
e.g. a strategy doesn't cleanly transfer all AUM to the new_strategy, but parent's `migrate()` transfers `want` anyway.
this `require` nudges strategists towards thinking about and explicitly defining what a successful migration means in their context.

Reply:
mmm as a strategist I would revert the tx if I can't cleanly transfer all AUM.
Do you think that having a `bool` return will force the behavior?

API decision for @fubuloubu

Reply:
> Do you think that having a bool return will force the behavior?

yes otherwise it should be an obvious flag for auditors, i.e. a yolo `return true` without any preceding `assert`ions